src/uppy-base/src/plugins/Provider.js

```diff
@@ -8,12 +8,12 @@ export default class Provider {
   constructor (opts) {
     this.opts = opts
     this.provider = opts.provider
-    this.id = this.provider
+    this.id = opts.id || this.provider
```

Comment:
I guess I could undo this and make each provider Plugin have an accessor method called `getAuthPath`(or something similar) instead. What do you think?